File: Python/save data/save_csv.py
# Save h2 headers to CSV file

from playwright.sync_api import sync_playwright
import csv

with sync_playwright() as p:
    browser = p.chromium.launch()
    page = browser.new_page()
    page.goto("https://example.com")
    headers = page.locator("h2").all_inner_texts()
    with open("headers.csv", "w", newline="", encoding="utf-8") as f:
        writer = csv.writer(f)
        writer.writerow(["Header"])
        for h in headers:
            writer.writerow([h])
    browser.close()

File: NodeJS/data/save_csv.js
// Save h2 headers to CSV file

const { chromium } = require('playwright');
const fs = require('fs');
const { parse } = require('json2csv');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  await page.goto('https://example.com');
  const headers = await page.$$eval('h2', els => els.map(e => e.innerText));
  const csv = parse(headers.map(h => ({ Header: h })));
  fs.writeFileSync('headers.csv', csv);
  await browser.close();
})();
